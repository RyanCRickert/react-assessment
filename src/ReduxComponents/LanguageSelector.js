import React from 'react';
import { connect } from "react-redux";
import { changeLanguage } from "./actions/options";

export class LanguageSelector extends React.Component {
  onLanguageChange = (e) => {
    this.props.changeLanguage(e.target.value);
  }

  render() {
    return (
      <div>
        <select
        className="App-language_selector"
        onChange={this.onLanguageChange}
        value={this.props.language}
      >
        <option>English</option>
        <option>Spanish</option>
        <option>French</option>
        <option>German</option>
        <option>Korean</option>
      </select>
    </div>
    );
  }
}

const mapStateToProps = (state) => ({
  language: state.language
})

const mapDispatchToProps = (dispatch) => ({
  changeLanguage: (language) => dispatch(changeLanguage(language))
})

export default connect(mapStateToProps, mapDispatchToProps)(LanguageSelector);