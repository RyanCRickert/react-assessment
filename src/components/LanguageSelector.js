import React from 'react';
import { AppConsumer } from "./AppProvider";

export default class LanguageSelector extends React.Component {
  render() {
    return (
      <AppConsumer>
        {(context) => (
          <div>
          <select
          className="App-language_selector"
          onChange={context.onChangeLanguage}
          value={context.state.language}
        >
          <option>English</option>
          <option>Spanish</option>
          <option>French</option>
          <option>German</option>
          <option>Korean</option>
        </select>
        </div>
        )}
    </AppConsumer>
    );
  }
}