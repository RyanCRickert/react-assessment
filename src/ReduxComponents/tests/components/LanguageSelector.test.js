import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { LanguageSelector } from "../../LanguageSelector";

let changeLanguage, wrapper;

beforeEach(() => {
  changeLanguage = jest.fn();
  wrapper =  shallow(
    <LanguageSelector
      changeLanguage={changeLanguage}
      />)
});

test("should render LanguageSelector", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should handle changing of language", () => {
  const value = "French";

  expect(toJSON(wrapper)).toMatchSnapshot();
  wrapper.find("select").simulate("change", {
    target: { value }
  });
  expect(changeLanguage).toHaveBeenCalled();
  expect(toJSON(wrapper)).toMatchSnapshot();
})