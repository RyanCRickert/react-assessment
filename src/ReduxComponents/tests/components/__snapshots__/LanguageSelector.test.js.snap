// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle changing of language 1`] = `
<div
  className="options-container"
>
  <select
    className="App-language_selector"
    onChange={[Function]}
  >
    <option>
      English
    </option>
    <option>
      Spanish
    </option>
    <option>
      French
    </option>
    <option>
      German
    </option>
    <option>
      Korean
    </option>
  </select>
  <button
    className="generator-button"
    onClick={[Function]}
  >
    Generate Names
  </button>
</div>
`;

exports[`should handle changing of language 2`] = `
<div
  className="options-container"
>
  <select
    className="App-language_selector"
    onChange={[Function]}
  >
    <option>
      English
    </option>
    <option>
      Spanish
    </option>
    <option>
      French
    </option>
    <option>
      German
    </option>
    <option>
      Korean
    </option>
  </select>
  <button
    className="generator-button"
    onClick={[Function]}
  >
    Generate Names
  </button>
</div>
`;

exports[`should render LanguageSelector 1`] = `
<div
  className="options-container"
>
  <select
    className="App-language_selector"
    onChange={[Function]}
  >
    <option>
      English
    </option>
    <option>
      Spanish
    </option>
    <option>
      French
    </option>
    <option>
      German
    </option>
    <option>
      Korean
    </option>
  </select>
  <button
    className="generator-button"
    onClick={[Function]}
  >
    Generate Names
  </button>
</div>
`;
